srcDoc/examples.rst

```diff
@@ -600,3 +600,359 @@ Maximum Throughput -- 2 *Routing Service*
+                DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                key);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "set_long(test_long) failed: %d.\n", retcode);
+            success = false;
+        }
+
+        if (snprintf(test_string, SIZE_TEST_STRING, "Hello World! %lu", key) < 0) {
+            success = false;
+        }
+        retcode = data.set_string(
+                "custom_type.test_string.test_string",
+                DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                test_string);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "set_string(test_string) failed: %d.\n", retcode);
+            success = false;
+        }
+
+        retcode = data.set_long(
+                "custom_type.test_enum",
+                DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                ENUM1);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "set_long(test_enum) failed: %d.\n", retcode);
+            success = false;
+        }
+
+        retcode = data.bind_complex_member(custom_type_data, "custom_type",
+                DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(
+                    stderr,
+                    "bind_complex_member(custom_type) failed: %d.\n",
+                    retcode);
+            success = false;
+        }
+        retcode = custom_type_data.bind_complex_member(test_seq_data, "test_seq",
+                DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(
+                    stderr,
+                    "bind_complex_member(test_seq_data) failed: %d.\n",
+                    retcode);
+            success = false;
+        }
+        retcode = test_seq_data.set_long_seq(
+                    "test_seq",
+                    DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                    long_seq);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "set_long(test_seq) failed: %d.\n", retcode);
+            success = false;
+        }
+        retcode = custom_type_data.unbind_complex_member(test_seq_data);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(
+                    stderr,
+                    "bind_complex_member(test_seq_data) failed: %d.\n",
+                    retcode);
+        }
+        retcode = data.unbind_complex_member(custom_type_data);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(
+                    stderr,
+                    "bind_complex_member(custom_type) failed: %d.\n",
+                    retcode);
+            success = false;
+        }
+        return success;
+    }
+
+- **finalize_custom_type_dynamic_data**:
+    This function is used to remove your data. It is called in the destructor.
+    The function takes one argument:
+
+        - A reference to the full DDS_DynamicData object including custom_type.
```

Comment:
either "object that includes the custom_type" or "object, including the custom_type," whichever you mean